OsmAnd/src/net/osmand/plus/chooseplan/ChoosePlanDialogFragment.java

```diff
@@ -352,18 +355,21 @@ private void setupOsmLiveCardButtons(boolean progress) {
 			osmLiveCardButtonsContainer.removeAllViews();
 			View lastBtn = null;
 			List<InAppSubscription> visibleSubscriptions = purchaseHelper.getLiveUpdates().getVisibleSubscriptions();
-			boolean anyPurchasedOrIntroducted = false;
+			boolean anyPurchased = false;
 			for (final InAppSubscription s : visibleSubscriptions) {
-				if (s.isPurchased() || s.getIntroductoryInfo() != null) {
-					anyPurchasedOrIntroducted = true;
+				if (s.isPurchased()) {
+					anyPurchased = true;
 					break;
 				}
 			}
+			InAppSubscription subscriptionMaxDiscount = purchaseHelper.getLiveUpdates()
+					.getSubscriptionWithMaxDiscount(purchaseHelper.getMonthlyLiveUpdates());
```

Comment:
Remove getSubscriptionWithMaxDiscount. First - it is wrong since it iterates through all and not visible. Second - it is redundant, use for loop above.